zerver/lib/email_notifications.py

```diff
@@ -569,9 +569,6 @@ def handle_missedmessage_emails(
     }
 
     user_profile = get_user_profile_by_id(user_profile_id)
-    if user_profile.is_bot:  # nocoverage # TODO -- needs a test.
-        # Never email bot users.
-        return
 
     if not user_profile.enable_offline_email_notifications:
         # BUG: Investigate why it's possible to get here.
```

Comment:
The theory for how we'd get here is just that the user configured notifications for the target stream, right?  Perhaps we can turn this into a check verifying that the user has configured notifications for the stream, and if not, log an error and return?  We can probably delete that code once it's been in production and confirmed our understanding of the world, but that'd let us fully close out this investigation.

Reply:
> The theory for how we'd get here is just that the user configured notifications for the target stream, right?

You mean the block with the `# BUG: Investigate why it's possible to get here.`, right? That shall be handled as part of fixing #19451. I'm only working only the "bots" case in this PR.

Reply:
I expect #19451 to be bunch of work. Not sure when I'll have the time for it, but I'm going to take that up next.